### PR TITLE
Fix for Redshift error analyzing metrics

### LINKED ISSUE #782: [Bug] Redshift error analyzing metrics
### Summary

When analyzing a metric, we do a UNION to combine an overall summary and a per-day breakdown.

```sql
SELECT
  null as date,
  o.*
FROM
  __overall o
UNION ALL
SELECT
  o.*
FROM
  __byDateOverall o
ORDER BY
  date ASC
```

In some instances, this causes the following error in Redshift:
> Query unsupported due to an internal error.

Moving this UNION ALL to a CTE seems to solve the problem:

```sql
WITH
  __union as (
    SELECT
      null as date,
      o.*
    FROM
      __overall o
    UNION ALL
    SELECT
      d.*
    FROM
      __byDateOverall d
  )
SELECT
  *
FROM
  __union
ORDER BY
  date ASC
```

### PR DESCRIPTION
**Pull request by Auz**
When analyzing a metric, the UNION to combine an overall summary and a per-day breakdown can cause an error with Redshift. Moved this UNION ALL to a CTE seems to prevent this problem.

Fixes #782

### CODE REVIEW

File: packages/back-end/src/integrations/SqlIntegration.ts [hunk 1]
```diff
@@ -408,23 +408,36 @@ export default abstract class SqlIntegration
...
         FROM
-          __byDateOverall o
+          __union
         ORDER BY
           date ASC
       `
-          : ""
+          : `
+        SELECT
+          o.*
```
Comment (jdorn) [inline packages/back-end/src/integrations/SqlIntegration.ts:435]: This is missing the `date` column.  We might have some code that depends on that existing, even if the value is just `null`.  I would add it back to be 100% backwards compatible.

Comment (Auz) [reply packages/back-end/src/integrations/SqlIntegration.ts:435]: Given our discussion, are we okay with this change now? The columns returned are the same as they were

**Comment by lukesonnet** [comment]
Both versions pass sqlfluff, and both work in my limited testing across postgres and mysql (no ability to test redshift itself).

File: packages/back-end/src/integrations/SqlIntegration.ts (hunk 1 above)
Comment (lukesonnet) [reply packages/back-end/src/integrations/SqlIntegration.ts:435]: I agree with Graham, there's no `date` column in the main branch if `params.includeByDate` is false.

**Comment by lukesonnet** [comment]
I think we should land this:
1. SQLFluff thinks both versions are fine for redshift, so at least we aren't worse on that metric.
2. The results for mySQL and Postgres are the same across versions. (see screenshots)
3. There's one fewer conditional, nice!
4. I assume that the original issue reporter or PR generator could actually test on Redshift and that this resolves the issue. But, I cannot confirm this myself.

Screenshots showing at least it doesn't change results in some cases (mysql):

Before:
Screen Shot 2023-02-17 at 2 13 05 PM

After:
Screen Shot 2023-02-17 at 2 12 22 PM

**Review by lukesonnet — APPROVED** [review]

File: packages/back-end/src/integrations/SqlIntegration.ts (hunk 1 above)
Comment (lukesonnet) [reply packages/back-end/src/integrations/SqlIntegration.ts:439]: Also, it looks like there's absolutely no code that calls the parent functions that build this SQL that set `includeByDate==false` so I can't even test it in the app, but the logic looks fine. Fixing/changing that seems out of scope.

**Review by jdorn — APPROVED** [review]

**Pull request merged**